src/org/zaproxy/zap/extension/ascanrulesAlpha/HttpOnlySite.java

```diff
@@ -0,0 +1,218 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import org.apache.commons.httpclient.URIException;
+import org.apache.commons.httpclient.URI;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.core.scanner.AbstractHostPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpStatusCode;
+import org.parosproxy.paros.network.HttpResponseHeader;
+import org.parosproxy.paros.Constant;
+
+import java.io.IOException;
+import java.net.SocketException;
+import java.net.SocketTimeoutException;
+import javax.net.ssl.SSLException;
+
+/**
+ * Active scan rule which raises an alert if a site accessed via HTTP is not served under HTTPS 
+ * https://github.com/zaproxy/zaproxy/issues/2207
+ * @author sanchitlucknow@gmail.com
+ */
+public class HttpOnlySite extends AbstractHostPlugin {
+
+	/**
+	 * Prefix for internationalised messages used by this rule
+	 */
+	private static final String MESSAGE_PREFIX = "ascanalpha.httponlysite.";
+	private static final int PLUGIN_ID = 10106;
+
+	private static final Logger log = Logger.getLogger(HttpOnlySite.class);
+	
+	@Override
+	public int getId() {
+		return PLUGIN_ID;
+	}
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "name");
+	}
+	
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return null;
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.MISC;
+	}
+	
+	@Override
+	public int getRisk() {
+		return Alert.RISK_MEDIUM;
+	}
+
+	@Override
+	public int getCweId() {
+		return 311; // CWE-311: Missing Encryption of Sensitive Data
+	}
+
+	@Override
+	public int getWascId() {
+		return 4; // WASC-04: Insufficient Transport Layer Protection
+	}
+
+	@Override
+	public void init() {
+
+	}
+
+	public void raiseAlert(HttpMessage newRequest, String message) {
+		String newUri = newRequest.getRequestHeader().getURI().toString();
+		bingo(getRisk(), //Risk
+				Alert.CONFIDENCE_MEDIUM, //Confidence/Reliability
+				getName(), //Name
+				getDescription(), //Description
+				getBaseMsg().getRequestHeader().getURI().toString(), //Original URI
+				null, //Param
+				"", //Attack
+				Constant.messages.getString(MESSAGE_PREFIX+"otherinfo."+message, newUri), //OtherInfo 
```

Comment:
I didn't mean to just use one (per detail) internationalised string, it can still use the main/detail strings as before, for example:
```Java
String otherInfoDetail = Constant.messages.getString(MESSAGE_PREFIX+"otherinfo." + message);
String otherInfo = Constant.messages.getString(MESSAGE_PREFIX+"otherinfo", otherInfoDetail, newUri);
```


```Properties
ascanalpha.httponlysite.otherinfo = {0}\nZAP attempted to connect via: {1}
ascanalpha.httponlysite.otherinfo.connectionfail = Failed to connect.
ascanalpha.httponlysite.otherinfo.differenthosts = Different Hosts.
```
It reduces the number of strings/words that need to be translated.